src/scopes/grants/startDate.js

```diff
@@ -1,26 +1,100 @@
 import { Op } from 'sequelize';
-import { withinDateRange, compareDate } from '../utils';
+import { compareDate } from '../utils';
 
-export function beforeGrantStartDate(date) {
+export function beforeStartDate(date) {
   return {
     [Op.and]: {
       [Op.or]: compareDate(date, 'startDate', Op.lt),
     },
   };
 }
 
-export function afterGrantStartDate(date) {
+export function afterStartDate(date) {
   return {
     [Op.and]: {
       [Op.or]: compareDate(date, 'startDate', Op.gt),
     },
   };
 }
 
-export function withinGrantStartDates(dates) {
+export function withinStartDates(dates) {
+  const scopes = dates.reduce((acc, range) => {
+    if (!range.split) {
+      return acc;
+    }
+
+    const [sd, ed] = range.split('-');
+    if (!sd || !ed) {
+      return acc;
+    }
+
+    return [
```

Comment:
I would not describe myself as incredibly confident that this is the best way to do this, it seems to pass my test cases but would love a second look at this

Reply:
Can we just check if the grant start date is before the passed in end date and the grant end date is after the passed in end date? I think I have all cases in the following screenshot, a grant can start before, during or after the date range. Any grants that start after the date range shouldn't be included. If a grant starts during a date range it is always included. If a grant starts before the date range it has to end after the start of the date range.
![Screenshot from 2022-01-26 11-38-03](https://user-images.githubusercontent.com/6631790/151216700-5f6e8f9b-4926-4a4e-a8b4-7e667eae2fa3.png)